pyxrf/model/draw_image_rgb.py

```diff
@@ -0,0 +1,353 @@
+                logger.warning('scaler data has zero values at {}'.format(np.where(self.scaler_data == 0)))
+                self.scaler_data[self.scaler_data == 0] = np.mean(self.scaler_data)
+                logger.warning('Use mean value {} instead for those points'.format(np.mean(self.scaler_data)))
+
+        if self.scale_opt == 'Linear':
+            for i, (k, v) in enumerate(six.iteritems(stat_temp)):
+
+                if self.scaler_data is not None:
+                    if k in name_not_scalable:
+                        data_dict = self.dict_to_plot[k]
+                    else:
+                        data_dict = self.dict_to_plot[k]/self.scaler_data*ic_norm
+
+                else:
+                    data_dict = self.dict_to_plot[k]
+
+                selected_data.append(data_dict)
+                selected_name.append(k) #self.file_name+'_'+str(k)
+
+        else:
+            for i, (k, v) in enumerate(six.iteritems(stat_temp)):
+
+                if self.scaler_data is not None:
+                    if k in name_not_scalable:
+                        data_dict = self.dict_to_plot[k]
+                    else:
+                        data_dict = self.dict_to_plot[k]/self.scaler_data*ic_norm
+
+                else:
+                    data_dict = np.log(self.dict_to_plot[k])
+
+                #maxz = np.max(data_dict)
+                selected_data.append(data_dict)
+                selected_name.append(k)
+
+        return selected_data, selected_name
+
+    def show_image(self):
+
+        selected_data, selected_name = self.preprocess_data()
+        selected_data = np.array(selected_data)
+
+        if len(selected_name) >0 and len(selected_name) < 3:
+            logger.error('Please select three elements for RGB plot.')
+        elif len(selected_name) >= 3:
+            if len(selected_name) > 3:
+                logger.warning('Please select only three elements for RGB plot.')
+            logger.warning('Elements {} are considered.'.format(selected_name[:3]))
+            self.rgb_name_list = selected_name[:3]
+
+        name_r, name_g, name_b = self.rgb_name_list
+
+        try:
+            data_r = selected_data[0,:,:]
+        except IndexError:
+            selected_data = np.ones([3,10,10])
+
+        name_r = self.rgb_name_list[self.index_red]
+        data_r = selected_data[self.index_red,:,:]
+        name_g = self.rgb_name_list[self.index_green]
+        data_g = selected_data[self.index_green,:,:]
+        name_b = self.rgb_name_list[self.index_blue]
+        data_b = selected_data[self.index_blue,:,:]
+
+        # norm data
+        data_r = norm_data(data_r)
+        data_g = norm_data(data_g)
+        data_b = norm_data(data_b)
+
+        R, G, B, RGB = make_cube(data_r,
+                                 data_g,
+                                 data_b)
+
+        self.fig.clf()
+
+        ax = self.fig.add_subplot(111)
+
+        ax_r, ax_g, ax_b = make_rgb_axes(ax, pad=0.02)
```

Comment:
Did you consider doing this in the __init__ and then keeping a reference to ax, ax_r, ax_g, ax_b so that you do not need to do this logic every time?